spec/rspec/core/reporter_spec.rb

```diff
@@ -172,6 +172,27 @@ module RSpec::Core
       end
     end
 
+    describe "#publish" do
+      let(:listener) { double("listener", :custom => nil) }
+      before do
+        reporter.register_listener listener, :custom, :start
+      end
+
+      it 'will send custom events to registered listeners' do
+        expect(listener).to receive(:custom).with(RSpec::Core::Notifications::NullNotification)
+        reporter.publish :custom
+      end
+      it 'will ignore RSpec standard events' do
+        reporter.publish :start
+      end
+      it 'will provide a custom notification object based on the options hash' do
+        expect(listener).to receive(:custom) do |notification|
+          expect(notification.my_data).to eq :value
+        end
```

Comment:
This could use `an_object_having_attributes`:

``` ruby
expect(listener).to receive(:custom).with(an_object_having_attributes(:my_data => :value))
```